examples/seq2seq_attn/seq2seq_attn.py

```diff
@@ -101,40 +106,39 @@ def forward(self, batch, mode):
             start_tokens = torch.ones_like(batch['target_length']) * \
                            self.bos_token_id
 
-            helper_infer = self.decoder.create_helper(
-                decoding_strategy="infer_greedy",
-                embedding=self.target_embedder,
+            infer_outputs = self.decoder(
                 start_tokens=start_tokens,
-                end_token=self.eos_token_id.item())
-
-            infer_outputs, _, _ = self.decoder(
-                helper=helper_infer,
-                memory=torch.cat(enc_outputs, dim=2),
+                end_token=self.eos_token_id.item(),
+                embedding=self.target_embedder,
+                memory=memory,
                 memory_sequence_length=batch['source_length'],
-                max_decoding_length=60)
+                beam_width=config_model.beam_width)
 
             return infer_outputs
 
 
 def main():
     """Entrypoint.
     """
-    train_data = tx.data.PairedTextData(hparams=config_data.train)
-    val_data = tx.data.PairedTextData(hparams=config_data.val)
-    test_data = tx.data.PairedTextData(hparams=config_data.test)
+    train_data = tx.data.PairedTextData(hparams=config_data.train,
+                                        device=device)
+    val_data = tx.data.PairedTextData(hparams=config_data.val,
+                                      device=device)
+    test_data = tx.data.PairedTextData(hparams=config_data.test,
+                                       device=device)
     data_iterator = tx.data.TrainTestDataIterator(
         train=train_data, val=val_data, test=test_data)
 
     model = Seq2SeqAttn(train_data)
-    optimizer = get_optimizer(model.parameters(), config_model.opt)
-    train_op = get_train_op(optimizer, config_model.opt)
+    model.to(device)
+    train_op = optimization.get_train_op(params=model.parameters(),
```

Comment:
I believe the intended usage suggested by @ZhitingHu was `tx.core.get_train_op`. Basically we only expose the first layer of packages to users.

Reply:
Yes, let's call it by `tx.core.get_train_op` as in the Texar TF examples